### ISSUE TITLE
Passing in two TensorDataset instances results in wrongly failed assertion

### ISSUE THREAD

**Issue by Nimmerwoner** [toshas/torch-fidelity#4]
Hi! Thanks for the package. While it looks exactly like what we need, we couldn't get it to work.

In our training loop, we call `torch_fidelity.calculate_metrics` as follows with two TensorDatasets made on the fly. We have verified their contents to be be torch.uint8 images, however we get the assertion error that the input is not torch.uint8. Investigating a little more, the cause seems to be that the input is a list instead.

```python
# in training loop
tensors1 = ... # image from network, already on GPU
tensors2 = ... # image from network, already on GPU
imgs1 = tensor_to_image(tensors1)  # changes BCWH to BWHC and converts to uint8
imgs2 = tensor_to_image(tensors2)  
dataset1 = TensorDataset(imgs1)
dataset2 = TensorDataset(imgs2)
frechet  = torch_fidelity.calculate_metrics(dataset1, dataset2, cuda=False, fid=True, save_cpu_ram=True)
```
Which results in:
```
File "/opt/conda/lib/python3.7/site-packages/torch_fidelity/feature_extractor_inceptionv3.py", line 81, in forward
    vassert(torch.is_tensor(x) and x.dtype == torch.uint8, 'Expecting image as torch.Tensor with dtype=torch.uint8')
File "/opt/conda/lib/python3.7/site-packages/torch_fidelity/helpers.py", line 9, in vassert
    raise ValueError(message)
ValueError: Expecting image as torch.Tensor with dtype=torch.uint8
```

Investigating, I found that it isn't `x.dtype == torch.uint8` which fails, but `torch.is_tensor(x)`. When dataset1 and dataset2 are created their inputs are all tensors, but somewhere within `calculate_metrics` these incorrectly get converted to lists. 

What's going wrong here and how can we fix this?

**Comment by toshas** [comment]
Hi, thanks for the report! Can you please insert the following lines before calling calculate_metrics and post here the outputs? There really seems no other reason to cause that exception..
```
print(type(dataset1), type(dataset2))
print(dataset1.dtype, dataset2.dtype)
```

**Comment by Nimmerwoner** [comment]
Hi, thanks for responding! It took us a while to get back to this, but we have added those lines and got the output (while still not working):

```
<class 'torch.utils.data.dataset.TensorDataset'> <class 'torch.utils.data.dataset.TensorDataset'>
torch.uint8 torch.uint8
```

We then tried making a minimal example script to illustrate the problem, but couldn't reproduce the error. Now we have it working, although we aren't quite sure what made the difference haha. We suspect we might have gone something wrong converting between different formats (normalized, colorized, BCWH, BWHC, ...). We now also use the Dataset class instead of TensorDataset.

Closing because the library works as-is.
